config/m4/compiler.m4

```diff
@@ -10,7 +10,20 @@
 #
 # Initialize CFLAGS
 #
-BASE_CFLAGS="-g -Wall -Werror"
+# BASE_CFLAGS sets compilation flags for GCC:
+# -g: Debug information
+# -Wall: All common warnings
+# -Werror: Treat warnings as errors
+# -Wframe-larger-than=8192: Warn if stack frame size exceeds 8192 bytes (8 KB)
+BASE_CFLAGS="-g -Wall -Werror -Wframe-larger-than=8192"
```

Comment:
might as well remove comments since the flags are so common

Reply:
removed "-Wall", "-Werror" and "-g"

Reply:
Outdated, done in other PRs